rfc/egress-billing.md

```diff
@@ -0,0 +1,309 @@
+# w3s.link gateway egress billing 
+
+## Authors
+
+- [Travis Vachon](https://github.com/travis)
+- [Irakli Gozalishvili](https://github.com/gozala)
+
+## Goals
+
+We need to start charging for data served by our gateway ("gateway egress"). Users who upload data using `w3up` should
+be charged when we serve that data through the gateway that is currently hosted at https://w3s.link. Importantly, if 
+two users upload the same data, we need to ensure the correct party is billed when we serve that data through
+the gateway - for instance, if I upload a file that a massive NFT platform also uploaded, I should not be charged
+when that platform serves that data to millions of their customers through their website.
+
+Charging for *all* reads is a non-goal - we believe there is a tradeoff between strict accounting and read performance
+and we generally want to err on the side of performant reads.
+
+## Abstract
+
+The business has the capacity to pay for some number of reads - we'll call this _R_. We will
+first implement a global per-CID rate-limit that is expected to account for N%
+of _R_. This rate limit will be used for all content that has not been configured with another
+rate limit, and for content that has been configured with other rate limits that have already
+been reached.
```

Comment:
It would be nice to model this as "just another user that set a rate limt", where "another user" is us. Eventually, some entity will have to pay the egress from the nodes storing the content, so we do need to be able to allocate responsibility for that egress to some entity.

i.e. there's no magic "global" limit, it's just some user (us) that has set a rate limit that happens to apply to ALL content served by the gateway.

Reply:
yea completely agree!

Reply:
🎯